pkg/api/api.go

```diff
@@ -156,10 +155,12 @@ func (s *Server) listAPIs(w http.ResponseWriter, r *http.Request) {
 
 	sort.Sort(apisByOrder(apiGroups))
 
-	buff, err := yaml.Marshal(apiGroups)
-	if err != nil {
-		panic(fmt.Errorf("marshal %#v to yaml failed: %v", apiGroups, err))
-	}
-	w.Header().Set("Content-Type", "text/vnd.yaml")
+	buff := spectool.MustMarshalJSON(apiGroups)
+
+	s.writeJSONBody(w, buff)
+}
+
+func (s *Server) writeJSONBody(w http.ResponseWriter, buff []byte) {
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
propose to check the `Accept` header to decide the response content type, though JSON is a subset of YAML technically if the user print out the response body directly, the format is very different and the user will be surprised if the request is YAML but the response is JSON.

Reply:
Theoretically, it's standard for RESTful APIs to only support json, and yaml is only for user interface level . That current APIs accept yaml is totally my backward compatibility concern.

I could support `Accept` header to support both formats, but it's not that standard. No strong opinion here, it's up to you.

Reply:
As we maintain the backward compatibility on the request side, I think maintaining the compatibility on the response side will be better.

Reply:
Then it is.

Reply:
Updated in system APIs. Mesh APIs remain json only because they have no need for backward compatibility.